src/measure_theory/integral/integral_eq_improper.lean

```diff
@@ -280,52 +280,80 @@ section integrable
 variables {α ι E : Type*} [measurable_space α] {μ : measure α} {l : filter ι}
   [normed_group E] [measurable_space E] [opens_measurable_space E]
 
+lemma ae_cover.integrable_of_lintegral_nnnorm_bounded [l.ne_bot] [l.is_countably_generated]
+  {φ : ι → set α} (hφ : ae_cover μ l φ) {f : α → E} (I : ℝ) (hfm : ae_measurable f μ)
+  (hbounded : ∀ᶠ i in l, ∫⁻ x in φ i, nnnorm (f x) ∂μ ≤ ennreal.of_real I) :
+  integrable f μ :=
+begin
+  refine ⟨hfm, (le_of_tendsto _ hbounded).trans_lt ennreal.of_real_lt_top⟩,
+  exact hφ.lintegral_tendsto_of_countably_generated
+    (measurable_nnnorm.comp_ae_measurable hfm).coe_nnreal_ennreal,
+end
+
 lemma ae_cover.integrable_of_lintegral_nnnorm_tendsto [l.ne_bot] [l.is_countably_generated]
   {φ : ι → set α} (hφ : ae_cover μ l φ) {f : α → E} (I : ℝ)
   (hfm : ae_measurable f μ)
   (htendsto : tendsto (λ i, ∫⁻ x in φ i, nnnorm (f x) ∂μ) l (𝓝 $ ennreal.of_real I)) :
   integrable f μ :=
 begin
-  refine ⟨hfm, _⟩,
-  unfold has_finite_integral,
-  rw hφ.lintegral_eq_of_tendsto _ (measurable_nnnorm.comp_ae_measurable hfm).coe_nnreal_ennreal
-    htendsto,
-  exact ennreal.of_real_lt_top
+  refine hφ.integrable_of_lintegral_nnnorm_bounded (max 1 (I + 1)) hfm _,
+  refine htendsto.eventually (ge_mem_nhds _),
+  refine (ennreal.of_real_lt_of_real_iff (lt_max_of_lt_left zero_lt_one)).2 _,
+  exact lt_max_of_lt_right (lt_add_one I),
 end
 
+lemma ae_cover.integrable_of_lintegral_nnnorm_bounded' [l.ne_bot] [l.is_countably_generated]
+  {φ : ι → set α} (hφ : ae_cover μ l φ) {f : α → E} (I : ℝ≥0) (hfm : ae_measurable f μ)
+  (hbounded : ∀ᶠ i in l, ∫⁻ x in φ i, nnnorm (f x) ∂μ ≤ ennreal.of_real I) :
```

Comment:
Do you really want to have a bound in terms of `ennreal.of_real I` instead of the coercion of `I` to `ennreal`?

Reply:
I'm happy either way! The original lemmas which this generalises uses that, so I went with it for consistency. Should I change both, or neither?

Reply:
If `I` is already `nnreal`, then we should use coercion.

Reply:
The coercion doesn't seem to work for me? Is there a coercion from reals to ennreals? I can't see one in `data/real/ennreal`

Reply:
Amended, thank you!